tests/app/main/test_frameworks.py

```diff
@@ -260,6 +255,222 @@ def test_last_updated_does_not_exist(self, data_api_client, s3):
 
             self._assert_last_updated_times(doc, last_updateds)
 
+    def test_returns_404_if_framework_does_not_exist(self, data_api_client, s3):
+        with self.app.test_client():
+            self.login()
+            data_api_client.get_framework.return_value = {'frameworks': {}}
```

Comment:
Not sure that this is the response you'd get from the API for a non-existent framework - I think it will 404.